oauth2/src/main/java/com/linecorp/armeria/client/auth/oauth2/AbstractOAuth2AuthorizationGrant.java

```diff
@@ -118,6 +99,84 @@ private static boolean shallObtainInsteadOfUpdate(Throwable throwable) {
      */
     @Override
     public final CompletionStage<GrantedOAuth2AccessToken> getAccessToken() {
-        return tokenManager.get();
+        final CompletableFuture<GrantedOAuth2AccessToken> future = new CompletableFuture<>();
+        final CompletableFuture<GrantedOAuth2AccessToken> tokenFuture = this.tokenFuture;
+
+        tokenFuture.handle((token, ignored) -> {
```

Comment:
If `tokenFuture` is completed exceptionally, we have to initiate another token issuing flow(the `future` here) and it shouldn't complete exceptionally inheriting the `cause` of `tokenFuture`.
It will never get a chance to update `tokenFuture` if we do this.

And expected behavior in `handle` callback is the same for both of the following cases:
* token == null && cause == null
* token == null && cause != null

So I've chosen to ignore the cause.